server/routes/money.js

```diff
@@ -38,6 +39,84 @@ const createMoneyRouter = ({ hubContentService, offenderService }) => {
     }
   });
 
+  router.get('/transactions', async (req, res, next) => {
+    try {
+      const config = {
+        content: false,
+        header: false,
+        postscript: true,
+        detailsType: 'small',
+        returnUrl: req.originalUrl,
+      };
+
+      const { user } = req;
+
+      const data = {};
+
+      if (user) {
+        const accountTypes = ['spends', 'private', 'savings'];
+        const accountType = accountTypes.includes(req.query.accountType)
+          ? req.query.accountType
+          : accountTypes[0];
+        const accountTypeMachineName = {
+          spends: 'spends',
+          private: 'cash',
+          savings: 'savings',
+        };
+
+        const transactionsTo = new Date();
+        const transactionsFrom = subDays(transactionsTo, 30);
+
+        const [transactions, balances] = await Promise.all([
+          offenderService.getTransactionsFor(
+            user,
+            accountTypeMachineName[accountType],
+            formatISO(transactionsFrom, { representation: 'date' }),
+            formatISO(transactionsTo, { representation: 'date' }),
+          ),
+          offenderService.getBalancesFor(user),
+        ]);
+
+        const listOfPrisons = Array.from(
+          new Set(transactions.map(transaction => transaction.prison)),
+        );
+        const prisonDetails = await Promise.all(
+          listOfPrisons.map(prisonId =>
+            offenderService.getPrisonDetailsFor(prisonId),
+          ),
+        );
+        const prisonDetailsLookup = prisonDetails.reduce(
+          (accumulator, prison) => {
+            if (Object.hasOwnProperty.call(accumulator, prison.prisonId)) {
+              return accumulator;
+            }
+            const updated = { ...accumulator };
+            updated[prison.prisonId] = prison.longDescription;
+            return updated;
+          },
+          {},
+        );
```

Comment:
Is `reduce` the most appropriate approach here? It feels like there's a lot of complexity in those lines, and I wonder if there's a more straightforward way of achieving the same outcome.